### PR TITLE
Add jest test infrastructure with 1 test

### PR DESCRIPTION
**Pull request by chamons**
- Jest configuration stolen with love from autorest core
- Fixes https://github.com/Azure/autorest.go/issues/670

### CODE REVIEW

**Comment by chamons** [comment]
The test is pretty trivial, but I wanted a "useful" test to go along with the infrastructure.

Once I get a green build, i'm going to break the test and verify CI fails.

**Comment by chamons** [comment]
Better:

```
> @autorest/go@4.0.0-preview.24 test /home/vsts/work/1/s/src
> jest --ci

FAIL test/parameter.test.ts
  parameter ordering
    ✕ required are before optional (5 ms)

  ● parameter ordering › required are before optional

    expect(received).toBe(expected) // Object.is equality

    Expected: false
    Received: true

      16 |         
      17 |         params.sort(sortParametersByRequired);
    > 18 |         expect(params[0].required).toBe(false);
         |                                    ^
      19 |         expect(params[1].required).toBe(true);
      20 |         expect(params[2].required).toBe(false);
      21 |         expect(params[3].required).toBe(false);

```

File: eng/pipelines/ci.yml [hunk 1]
```diff
@@ -128,6 +128,12 @@ jobs:
           popd
         displayName: 'Run Acceptance Tests'
 
+      - pwsh: |
+          pushd ./src
+          npm test
+          popd
+        displayName: 'Run Unit Tests'
```
Comment (benbp) [inline eng/pipelines/ci.yml:135]: `popd` is unnecessary here as shell context doesn't carry from step to step. You could also configure the shell directory externally if you want:

```
      - pwsh: npm test
        displayName: 'Run Unit Tests'
        workingDirectory: $(System.DefaultWorkingDirectory)/src
```

**Review by benbp — APPROVED** [review]

**Comment by chlowell** [comment]
@jhendrixMSFT this looks good to me, what do you think?

File: src/jest.config.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        tsconfig: "tsconfig.json",
+      },
+    },
+    setupFilesAfterEnv: [],
+    verbose: true,
+    testEnvironment: "node",
+    testMatch: ["<rootDir>/src/**/*.test.ts", "<rootDir>/test/**/*.test.ts"],
+};
+  
+module.exports = config;
```
Comment (chlowell) [inline src/jest.config.js:25]:
```suggestion
module.exports = config;

```

File: tsconfig.json [hunk 3]
```diff
@@ -17,13 +17,15 @@
...
     "lib": [
       "es2018", 
       "dom",
     ],
     "experimentalDecorators": true,
     "newLine": "LF",
-  }
+  },
+  "include": ["src/**/*.ts", "definitions/**/*.d.ts", "test/**/*.ts"]
 }
```
Comment (chlowell) [inline tsconfig.json:31]:
```suggestion
  "include": ["src/**/*.ts", "definitions/**/*.d.ts", "test/**/*.ts"]
}

```

**Review by chlowell — APPROVED** [review]
Looks good to me. Adding a couple suggestions to see whether I can use them to add terminal newlines.

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**